support/sier-codec/src/lib.rs

```diff
@@ -48,13 +52,17 @@ impl Parser {
     pub fn struct_def(&self, name: &str) -> Option<&Arc<StructDef>> {
         self.structs.values().find(|s| s.type_name() == name)
     }
-}
 
-impl Default for Parser {
-    fn default() -> Self {
-        Parser {
-            structs: HashMap::new(),
-        }
+    pub fn parse_json<'a>(
+        &'a self,
+        file_json_contents: &str,
```

Comment:
I think the name should be `json_str` since it doesn't necessarily need to be a file. And parameter names are included in documentation, so should be very high quality.